include/common/polymorphic_execution.h

```diff
@@ -0,0 +1,105 @@
+/**
+* @version		GrPPI v0.2
+* @copyright		Copyright (C) 2017 Universidad Carlos III de Madrid. All rights reserved.
+* @license		GNU/GPL, see LICENSE.txt
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You have received a copy of the GNU General Public License in LICENSE.txt
+* also available in <http://www.gnu.org/licenses/gpl.html>.
+*
+* See COPYRIGHT.txt for copyright notices and details.
+*/
+
+#ifndef GRPPI_POLYMORPHIC_EXECUTION_H
+#define GRPPI_POLYMORPHIC_EXECUTION_H
+
+#include <typeinfo>
+#include <memory>
+
+#include "common/seq_policy.h"
+#include "common/thread_policy.h"
+#include "common/tbb_policy.h"
+#include "common/omp_policy.h"
+
+namespace grppi{
+
+/// Meta-function to determine if a type is an execution policy
+template <typename E>
+constexpr bool is_execution_policy() {
+  return std::is_same<E, grppi::sequential_execution>::value 
+      || std::is_same<E, grppi::parallel_execution_thr>::value
+      || std::is_same<E, grppi::parallel_execution_tbb>::value
```

Comment:
The execution policy type will not exists if the system does not support a given framework. I propose two different solutions to this issue. The first solution is to wrap each type with the corresponding "ifdef". The second solution is modify the policies code to define empty classes for those frameworks that are not supported.